core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -643,7 +643,7 @@ public ShareAcquiredRecords acquire(
                         // acquire subset of offsets from the in-flight batch but only if the
                         // complete batch is available yet. Hence, do a pre-check to avoid exploding
                         // the in-flight offset tracking unnecessarily.
-                        if (inFlightBatch.batchState() != RecordState.AVAILABLE) {
+                        if (inFlightBatch.batchState() != RecordState.AVAILABLE || inFlightBatch.batchRollbackState() == RecordState.ACQUIRED) {
```

Comment:
I think that really it's the presence of a rollback state at all, and not whether it would roll back to ACQUIRED which is interesting. The test purpose of the complete test is to see whether the batch is unencumbered.

Reply:
Hi @AndrewJSchofield , although the original change would have worked because we only do a commit from `ACQUIRED -> ACKNOWLEDGED` or `ACQUIRED -> AVAILABLE` or `ACQUIRED -> ARCHIVED`, but I think the change to check from ACQUIRED to null will help in future code extension, hence made the change as adviced. Same for the below 2 comments as well

Reply:
Indeed. That was entirely why I made the comment. Thanks :)